lib/Doctrine/DBAL/SQLParserUtils.php

```diff
@@ -250,11 +251,12 @@ private static function getUnquotedStatementFragments($statement)
             self::ESCAPED_DOUBLE_QUOTED_TEXT . '|' .
             self::ESCAPED_BACKTICK_QUOTED_TEXT . '|' .
             self::ESCAPED_BRACKET_QUOTED_TEXT;
-        $expression = sprintf('/((.+(?i:ARRAY)\\[.+\\])|([^\'"`\\[]+))(?:%s)?/s', $literal);
-
+        $expression = sprintf('/(?:[\s]*--.*?\n)|((.+(?i:ARRAY)\\[.+\\])|([^-\'"`\\[]+))(?:%s)?/s', $literal);
         preg_match_all($expression, $statement, $fragments, PREG_OFFSET_CAPTURE);
 
-        return $fragments[1];
+        return array_filter($fragments[1], static function ($match) {
```

Comment:
Please use argument and return value types where possible (`function (string $match) : bool`).